### PR TITLE
added first version application

### PR DESCRIPTION
**Pull request by Fedich09**

### CODE REVIEW

File: src/main/java/mate/Application.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private static Injector injector = Injector
+            .getInstance("mate");
+
+    public static void main(String[] args) {
+        ManufacturerService serviceManufacturer =
+                (ManufacturerService) injector.getInstance(ManufacturerService.class);
+        Manufacturer manufacturer = new Manufacturer("BMW", "Germany");
+        Manufacturer manufacturer1 = new Manufacturer("Mercedes", "Germany");
+        Manufacturer manufacturer2 = new Manufacturer("Mitsubishi", "Japan");
+
```
Comment (AndreiYazadzhi) [inline src/main/java/mate/Application.java]: Empty line

File: src/main/java/mate/Application.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        serviceManufacturer.add(manufacturer);
+        serviceManufacturer.add(manufacturer2);
+
```
Comment (AndreiYazadzhi) [inline src/main/java/mate/Application.java]: Empty line

File: src/main/java/mate/Application.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        serviceManufacturer.update(1L, manufacturer1);
+        serviceManufacturer.deleteById(1L);
+        serviceManufacturer.add(manufacturer);
+        serviceManufacturer.delete(manufacturer);
+        List<Manufacturer> all = serviceManufacturer.getAll();
+        System.out.println(all.toString());
+        System.out.println();
+
```
Comment (AndreiYazadzhi) [inline src/main/java/mate/Application.java]: Empty line

File: src/main/java/mate/db/Storage.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import mate.model.Manufacturer;
+
+public class Storage {
+    public static List<Manufacturer> manufacturers = new ArrayList<>();
+    private static Long manufactureId = 0L;
+    private static Long driversId = 0L;
+    private static Long carsId = 0L;
```
Comment (AndreiYazadzhi) [inline src/main/java/mate/db/Storage.java]: Redundant variables

File: src/main/java/mate/model/Car.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car() {
+    }
+
+    public Car(String model, Manufacturer manufacturer) {
+        this.model = model;
+        this.manufacturer = manufacturer;
+        drivers = new ArrayList<>();
```
Comment (AndreiYazadzhi) [inline src/main/java/mate/model/Car.java:15]: Why we create new List every time when create Car?

**Review by AndreiYazadzhi — CHANGES_REQUESTED** [review]

**Comment by AndreiYazadzhi** [comment]
Fix your Travis

File: src/main/java/mate/model/Car.java (hunk 5 above)
Comment (mfedenko09) [reply src/main/java/mate/model/Car.java:15]: Untitled

Comment (Fedich09) [reply src/main/java/mate/model/Car.java:18]: Untitled

**Review by AndreiYazadzhi — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -9,8 +9,52 @@
...
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
 </project>
```
Comment (Sofasmile) [inline pom.xml]: let's add an empty line at the end of the file

File: src/main/java/mate/dao/Dao.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+public interface Dao<T> {
+    T add(T object);
+
+    Optional<T> getById(Long id);
+
+    T update(Long id, T object);
```
Comment (Sofasmile) [inline src/main/java/mate/dao/Dao.java]:
```suggestion
    T update(T object);
```

File: src/main/java/mate/dao/impl/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        return Storage.manufacturers.stream()
+                .filter(e -> e.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public Manufacturer update(Long id, Manufacturer manufacturer) {
+        Manufacturer old = Storage.manufacturers.get(id.intValue());
+        manufacturer.setId(id);
+        Storage.manufacturers.set(id.intValue() - 1, manufacturer);
```
Comment (Sofasmile) [inline src/main/java/mate/dao/impl/ManufacturerDaoImpl.java]: does is work correctly? 
lets' test this method with a lot of data:
```
create(new Manufacturer("bmw1", "UK1")); // id = 1
create(new Manufacturer("bmw2", "UK2")); // id = 2
create(new Manufacturer("bmw3", "UK3")); // id = 3
create(new Manufacturer("bmw4", "UK4")); // id = 4
deleteById(1L);
deleteById(2L);
update(...) // let's update last manufacturer -  "id = 4", "name = bmw4", "country = UK4" and change country for example to "UK5"
```

File: src/main/java/mate/dao/impl/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        return old;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        for (int i = 0; i < Storage.manufacturers.size(); i++) {
```
Comment (Sofasmile) [inline src/main/java/mate/dao/impl/ManufacturerDaoImpl.java]: read about method `removeIf`

File: src/main/java/mate/dao/impl/ManufacturerDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean delete(Manufacturer manufacturer) {
+        for (int i = 0; i < Storage.manufacturers.size(); i++) {
+            if (Storage.manufacturers.get(i).equals(manufacturer)) {
+                Storage.manufacturers.remove(manufacturer);
```
Comment (Sofasmile) [inline src/main/java/mate/dao/impl/ManufacturerDaoImpl.java]: what does method remove() return?

File: src/main/java/mate/model/Car.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class Car {
+    private Long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car() {
```
Comment (Sofasmile) [inline src/main/java/mate/model/Car.java]: do we need empty constructor?

File: src/main/java/mate/service/Service.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
+package mate.service;
+
+import java.util.List;
+
+public interface Service<T> {
+    T add(T object);
+
+    T getById(Long t);
+
+    T update(Long id, T object);
```
Comment (Sofasmile) [inline src/main/java/mate/service/Service.java]:
```suggestion
    T update(T object);
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/mate/dao/impl/ManufacturerDaoImpl.java [hunk 13]
```diff
@@ -22,33 +22,25 @@ public Manufacturer add(Manufacturer manufacturer) {
...
-        Storage.manufacturers.set(id.intValue() - 1, manufacturer);
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer old = new Manufacturer();
+        for (int i = 0; i < Storage.manufacturers.size(); i++) {
+            if (manufacturer.getId().equals(Storage.manufacturers.get(i).getId())) {
+                old = Storage.manufacturers.get(i);
+                Storage.manufacturers.set(i, manufacturer);
+            }
+        }
         return old;
```
Comment (Sofasmile) [inline src/main/java/mate/dao/impl/ManufacturerDaoImpl.java]: we shouldn't return old value, let's return new manufacturer

File: src/main/java/mate/dao/impl/ManufacturerDaoImpl.java [hunk 14]
```diff
@@ -22,33 +22,25 @@ public Manufacturer add(Manufacturer manufacturer) {
...
     @Override
     public boolean delete(Manufacturer manufacturer) {
-        for (int i = 0; i < Storage.manufacturers.size(); i++) {
-            if (Storage.manufacturers.get(i).equals(manufacturer)) {
-                Storage.manufacturers.remove(manufacturer);
-                return true;
-            }
-        }
-        return false;
+        return Storage.manufacturers.removeIf(e -> e.equals(manufacturer));
```
Comment (Sofasmile) [inline src/main/java/mate/dao/impl/ManufacturerDaoImpl.java]:
```suggestion
        return Storage.manufacturers.remove(manufacturer);
```

File: src/main/java/mate/model/Car.java [hunk 15]
```diff
@@ -0,0 +1,28 @@
+package mate.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Car {
+    private Long id;
```
Comment (Sofasmile) [inline src/main/java/mate/model/Car.java:7]: let's add getters and setters

File: src/main/java/mate/model/Driver.java [hunk 16]
```diff
@@ -0,0 +1,20 @@
+package mate.model;
+
+public class Driver {
+    private Long id;
```
Comment (Sofasmile) [inline src/main/java/mate/model/Driver.java:4]: let's add getters and setters

File: src/main/java/mate/model/Manufacturer.java [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+    private Long id;
+    private String name;
+    private String country;
+
+    public Manufacturer(String name, String country) {
+        this.name = name;
+        this.country = country;
+    }
+
+    public Manufacturer() {
```
Comment (Sofasmile) [inline src/main/java/mate/model/Manufacturer.java]: do we need empty constructor?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**